### ISSUE TITLE
Invalid rule configuration only gives feedback on console and does not update UI at all

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint.github.io#413]
Repro steps:

1. Enter the following into the demo:
    ```js
    /* eslint no-unreachable: ["error"] */
    function foo() {
        return;
        doSomething();
    }
    ```
1. Observe no-unreachable error
1. Change configuration to: `/* eslint no-unreachable: ["error", { someInvalidConfig: true }] */`
1. Remove the `return;` statement

Expected: Get an error about invalid rule configuration at some point

Actual: Previous error is stuck on the screen

Note: This also applies to lint-free code, and code will appear to remain "lint-free" even if there are actually lint problems.
